src/data_cell/flatten_interface.h

```diff
@@ -33,38 +46,57 @@ class FlattenInterface {
     Train(const float* data, uint64_t count) = 0;
 
     virtual void
-    InsertVector(const float* vector, uint64_t idx = INT64_MAX) = 0;
+    InsertVector(const float* vector,
+                 InnerIdType idx = std::numeric_limits<InnerIdType>::max()) = 0;
 
     virtual void
-    BatchInsertVector(const float* vectors, uint64_t count, uint64_t* idx = nullptr) = 0;
+    BatchInsertVector(const float* vectors, InnerIdType count, InnerIdType* idx = nullptr) = 0;
 
     virtual float
-    ComputePairVectors(uint64_t id1, uint64_t id2) = 0;
+    ComputePairVectors(InnerIdType id1, InnerIdType id2) = 0;
+
+    virtual void
+    Prefetch(InnerIdType id) = 0;
 
 public:
     virtual void
-    SetMaxCapacity(uint64_t capacity) {
+    SetMaxCapacity(InnerIdType capacity) {
         this->max_capacity_ = capacity;
     };
 
     [[nodiscard]] virtual const uint8_t*
-    GetCodesById(uint64_t id, bool& need_release) const {
+    GetCodesById(InnerIdType id, bool& need_release) const {
         return nullptr;
     }
 
     virtual bool
-    GetCodesById(uint64_t id, uint8_t* codes) const {
+    GetCodesById(InnerIdType id, uint8_t* codes) const {
         return false;
     }
 
-    [[nodiscard]] virtual uint64_t
+    [[nodiscard]] virtual InnerIdType
     TotalCount() const {
         return this->total_count_;
     }
 
+    virtual void
+    Serialize(StreamWriter& writer) {
+        StreamWriter::WriteObj(writer, this->total_count_);
+        StreamWriter::WriteObj(writer, this->max_capacity_);
+        StreamWriter::WriteObj(writer, this->code_size_);
+    }
+
+    virtual void
+    Deserialize(StreamReader& reader) {
+        StreamReader::ReadObj(reader, this->total_count_);
+        StreamReader::ReadObj(reader, this->max_capacity_);
+        StreamReader::ReadObj(reader, this->code_size_);
+    }
+
 public:
-    uint64_t total_count_{0};
-    uint64_t max_capacity_{1000000};
-    uint64_t code_size_{0};
+    InnerIdType total_count_{0};
+    InnerIdType max_capacity_{1000000};
+    uint32_t code_size_{0};
 };
+
```

Comment:
please add an alias here, `using FlattenInterfacePtr = std::shared_ptr<FlattenInterface>`, and use it other where

Reply:
done